test/extended/networking/commatrix.go

```diff
@@ -0,0 +1,158 @@
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var _ = g.Describe("[sig-network][Feature:commatrix][Serial]", func() {
+	g.It("should apply firewall by blocking all ports except the ones OCP is listening on", func() {
+		g.By("generating the commatrix")
+		kubeconfig, ok := os.LookupEnv("KUBECONFIG")
+		if !ok {
+			g.Fail("must set the KUBECONFIG environment variable")
+		}
+
+		cs, err := clientutil.New(kubeconfig)
+		o.Expect(err).ToNot(o.HaveOccurred())
+
+		deployment := commatrix.MNO
+		isSNO, err := isSNOCluster(cs.ConfigV1Interface)
+		o.Expect(err).NotTo(o.HaveOccurred())
+		if isSNO {
+			deployment = commatrix.SNO
+		}
+
+		mat, err := commatrix.New(kubeconfig, "", "", commatrix.Cloud, deployment)
+		o.Expect(err).ToNot(o.HaveOccurred())
+
+		masterMat, workerMat := separateMatrixByRole(*mat)
+
+		masterNFT, err := types.ToNFTables(masterMat)
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		workerNFT, err := types.ToNFTables(workerMat)
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		ns := consts.DefaultDebugNamespace + "firewall"
+		err = debug.CreateNamespace(cs, ns)
+		o.Expect(err).ToNot(o.HaveOccurred())
+
+		defer func() {
+			err := debug.DeleteNamespace(cs, ns)
+			o.Expect(err).ToNot(o.HaveOccurred())
+		}()
+
+		g.By("apply firewall on master nodes")
+		err = applyRulesToNode(cs, "master", ns, masterNFT)
+		o.Expect(err).ToNot(o.HaveOccurred())
+
+		if !isSNO {
+			g.By("apply firewall on worker nodes")
+			err = applyRulesToNode(cs, "worker", ns, workerNFT)
+			o.Expect(err).ToNot(o.HaveOccurred())
+		}
+	})
+})
+
+func applyRulesToNode(cs *clientutil.ClientSet, role, ns string, NFTtable []byte) error {
+	nodesList, err := cs.CoreV1Interface.Nodes().List(context.TODO(), metav1.ListOptions{})
+	if err != nil {
+		return fmt.Errorf("failed to list nodes: %w", err)
+	}
+
+	for _, node := range nodesList.Items {
+		if nodesutil.GetRole(&node) != role {
+			continue
+		}
+
+		debugPod, err := debug.New(cs, node.Name, ns, consts.DefaultDebugPodImage)
+		if err != nil {
+			return fmt.Errorf("failed to create debug pod on node %s: %w", node.Name, err)
+		}
+
+		defer func() {
+			err := debugPod.Clean()
+			if err != nil {
+				fmt.Printf("failed cleaning debug pod %s: %v", debugPod, err)
+			}
+		}()
+
+		command := fmt.Sprintf("echo '%s' > %s", string(NFTtable), "firewall.nft")
+		_, err = debugPod.Exec(command)
+		if err != nil {
+			return fmt.Errorf("failed to save rule set to file on node %s: %w", node.Name, err)
+		}
+
+		command = "nft -f firewall.nft "
+		_, err = debugPod.Exec(command)
+		if err != nil {
+			return fmt.Errorf("failed to apply rule set %q on node %s: %w", command, node.Name, err)
+		}
+
+		// Check the output of nft list ruleset
+		checkCommand := "nft list ruleset"
+		output, err := debugPod.Exec(checkCommand)
+		if err != nil {
+			return fmt.Errorf("failed to list NFT ruleset on node %s: %w", node.Name, err)
+		}
+
+		if len(output) == 0 {
```

Comment:
This doesn't actually validating the rules you applied since this output is never empty